luigi/contrib/bigquery.py

```diff
@@ -355,7 +363,14 @@ def run(self):
         if self.schema:
             job['configuration']['load']['schema'] = {'fields': self.schema}
 
-        bq_client.run_job(output.table.project_id, job, dataset=output.table.dataset)
+        logger.info('Loading data to temporary table %s.%s', output.tmp_table.dataset_id, output.tmp_table.table_id)
+        bq_client.run_job(output.tmp_table.project_id, job, dataset=output.tmp_table.dataset)
+        logger.info('Moving temporary table %s.%s to destination table %s.%s',
+                    output.tmp_table.dataset_id, output.tmp_table.table_id,
+                    output.table.dataset_id, output.table.table_id)
+        bq_client.copy(output.tmp_table, output.table)
```

Comment:
Yes, I believe it's just a reassignment of ids on googles side.  IIRC there is no actual copy of data involved, and [table operations are all atomic](http://stackoverflow.com/a/12514790/215608)